cluster/service.go

```diff
@@ -51,74 +45,78 @@ type Stats struct {
 	CacheRefreshes int
 }
 
-// NewService creates a Resolver that rely on the Auth service to retrieve tokens
-func NewService(authURL string, clustersRefreshDelay time.Duration, serviceToken string, resolveToken token.Resolve, decode token.Decode, options ...configuration.HTTPClientOption) (Service, error) {
+type clusterService struct {
+	authService      *auth.Service
+	cacheRefresher   *time.Ticker
+	cacheRefreshLock *sync.RWMutex
+	cacheHits        int
+	cacheMissed      int
+	cacheRefreshes   int
+	cachedClusters   []Cluster
+}
+
+// NewClusterService creates an instance of service that using the Auth service retrieves information about clusters
+func NewClusterService(refreshInt time.Duration, authService *auth.Service) Service {
 	// setup a ticker to refresh the cluster cache at regular intervals
-	cacheRefresher := time.NewTicker(clustersRefreshDelay)
-	s := &clusterService{
-		authURL:          authURL,
-		serviceToken:     serviceToken,
-		resolveToken:     resolveToken,
-		decode:           decode,
+	cacheRefresher := time.NewTicker(refreshInt)
+	service := &clusterService{
+		authService:      authService,
 		cacheRefresher:   cacheRefresher,
 		cacheRefreshLock: &sync.RWMutex{},
-		clientOptions:    options}
-	// immediately load the list of clusters before returning
+	}
+	return service
+}
+
+func (s *clusterService) Start() error {
+	//immediately load the list of clusters before returning
 	err := s.refreshCache(context.Background())
 	if err != nil {
-		log.Error(nil, map[string]interface{}{"error_message": err}, "failed to load the list of clusters during service initialization")
-		return nil, err
+		return fmt.Errorf("failed to load the list of clusters during service initialization: %s", err)
 	}
 	go func() {
-		for range cacheRefresher.C { // while the `cacheRefresh` ticker is running
+		for range s.cacheRefresher.C { // while the `cacheRefresh` ticker is running
 			s.refreshCache(context.Background())
```

Comment:
Log the error here. Don't just ignore it.

Reply:
blindly copied from the original code :-).
fixed in 5d94d71

Reply:
Still not fixed :)

Reply:
what the hell? for some reason, this [commit](14c8dba) removed the part that I added before... probably result of some crazy rebasing/stashing...
fixed again in: 008645a 
sorry for that and thanks for finding it